common/lib/xmodule/xmodule/mongo_utils.py

```diff
@@ -34,6 +34,9 @@ def connect_to_mongodb(
         # No 'replicaSet' in kwargs - so no secondary reads.
         mongo_client_class = pymongo.MongoClient
 
+    # If the MongoDB server uses a separate authentication database that should be specified here
+    auth_source = kwargs.pop('auth_source', '') or None
```

Comment:
@xitij2000, couldn't this be written using the default value parameter of the `.pop()` method?
```suggestion
    auth_source = kwargs.pop('auth_source', None)
```